server/loginserver.ts

```diff
@@ -76,23 +74,25 @@ class LoginServerInstance {
 		const actionUrl = url.parse(this.uri + 'action.php' +
 			'?act=' + action + '&serverid=' + Config.serverid +
 			'&servertoken=' + encodeURIComponent(Config.servertoken) +
-			'&nocache=' + new Date().getTime() + dataString);
-
-		return new Promise((resolve, reject) => {
-			const req = http.get(actionUrl, (res: IncomingMessage) => {
-				void Streams.readAll(res).then((buffer: string) => {
-					const result = parseJSON(buffer).json || null;
-					resolve([result, res.statusCode || 0, null]);
-					this.openRequests--;
+			'&nocache=' + new Date().getTime() + dataString).href;
+
+		return new Promise((resolve) => {
```

Comment:
Do you think you can refactor this into async/await for me?

Reply:
I can try, sure.